### ISSUE TITLE
Lots of heavy queries on MySQL

### ISSUE THREAD

**Issue by samvignoli** [rancher/rancher#3013]
Any ideias on how we can get rid of this?

Thanks.

```sql
mysql> show full processlist;
| 647 | cattle | localhost:59078 | cattle | Query   |  470 | Copying to tmp table | select `nic`.`id`, `nic`.`name`, `nic`.`account_id`, `nic`.`kind`, `nic`.`uuid`, `nic`.`description`, `nic`.`state`, `nic`.`created`, `nic`.`removed`, `nic`.`remove_time`, `nic`.`data`, `nic`.`instance_id`, `nic`.`network_id`, `nic`.`subnet_id`, `nic`.`vnet_id`, `nic`.`device_number`, `nic`.`mac_address` from `nic` where `nic`.`id` in (select max(`nic`.`id`) from `nic` join `vnet` on `vnet`.`id` = `nic`.`vnet_id` join `instance` on `instance`.`id` = `nic`.`instance_id` where (`nic`.`account_id` = 5 and `nic`.`removed` is null and `instance`.`removed` is null and `instance`.`system_container` is null) group by `nic`.`vnet_id`) |
| 648 | cattle | localhost:59079 | cattle | Query   |  470 | Copying to tmp table | select `nic`.`id`, `nic`.`name`, `nic`.`account_id`, `nic`.`kind`, `nic`.`uuid`, `nic`.`description`, `nic`.`state`, `nic`.`created`, `nic`.`removed`, `nic`.`remove_time`, `nic`.`data`, `nic`.`instance_id`, `nic`.`network_id`, `nic`.`subnet_id`, `nic`.`vnet_id`, `nic`.`device_number`, `nic`.`mac_address` from `nic` where `nic`.`id` in (select max(`nic`.`id`) from `nic` join `vnet` on `vnet`.`id` = `nic`.`vnet_id` join `instance` on `instance`.`id` = `nic`.`instance_id` where (`nic`.`account_id` = 5 and `nic`.`removed` is null and `instance`.`removed` is null and `instance`.`system_container` is null) group by `nic`.`vnet_id`) |
| 649 | cattle | localhost:59081 | cattle | Query   |  470 | Copying to tmp table | select `nic`.`id`, `nic`.`name`, `nic`.`account_id`, `nic`.`kind`, `nic`.`uuid`, `nic`.`description`, `nic`.`state`, `nic`.`created`, `nic`.`removed`, `nic`.`remove_time`, `nic`.`data`, `nic`.`instance_id`, `nic`.`network_id`, `nic`.`subnet_id`, `nic`.`vnet_id`, `nic`.`device_number`, `nic`.`mac_address` from `nic` where `nic`.`id` in (select max(`nic`.`id`) from `nic` join `vnet` on `vnet`.`id` = `nic`.`vnet_id` join `instance` on `instance`.`id` = `nic`.`instance_id` where (`nic`.`account_id` = 5 and `nic`.`removed` is null and `instance`.`removed` is null and `instance`.`system_container` is null) group by `nic`.`vnet_id`) |
| 650 | cattle | localhost:59080 | cattle | Query   |  470 | Copying to tmp table | select `nic`.`id`, `nic`.`name`, `nic`.`account_id`, `nic`.`kind`, `nic`.`uuid`, `nic`.`description`, `nic`.`state`, `nic`.`created`, `nic`.`removed`, `nic`.`remove_time`, `nic`.`data`, `nic`.`instance_id`, `nic`.`network_id`, `nic`.`subnet_id`, `nic`.`vnet_id`, `nic`.`device_number`, `nic`.`mac_address` from `nic` where `nic`.`id` in (select max(`nic`.`id`) from `nic` join `vnet` on `vnet`.`id` = `nic`.`vnet_id` join `instance` on `instance`.`id` = `nic`.`instance_id` where (`nic`.`account_id` = 5 and `nic`.`removed` is null and `instance`.`removed` is null and `instance`.`system_container` is null) group by `nic`.`vnet_id`) |
| 651 | cattle | localhost:59084 | cattle | Query   |  470 | Copying to tmp table | select `nic`.`id`, `nic`.`name`, `nic`.`account_id`, `nic`.`kind`, `nic`.`uuid`, `nic`.`description`, `nic`.`state`, `nic`.`created`, `nic`.`removed`, `nic`.`remove_time`, `nic`.`data`, `nic`.`instance_id`, `nic`.`network_id`, `nic`.`subnet_id`, `nic`.`vnet_id`, `nic`.`device_number`, `nic`.`mac_address` from `nic` where `nic`.`id` in (select max(`nic`.`id`) from `nic` join `vnet` on `vnet`.`id` = `nic`.`vnet_id` join `instance` on `instance`.`id` = `nic`.`instance_id` where (`nic`.`account_id` = 5 and `nic`.`removed` is null and `instance`.`removed` is null and `instance`.`system_container` is null) group by `nic`.`vnet_id`) |
| 652 | cattle | localhost:59083 | cattle | Query   |  470 | Copying to tmp table | select `nic`.`id`, `nic`.`name`, `nic`.`account_id`, `nic`.`kind`, `nic`.`uuid`, `nic`.`description`, `nic`.`state`, `nic`.`created`, `nic`.`removed`, `nic`.`remove_time`, `nic`.`data`, `nic`.`instance_id`, `nic`.`network_id`, `nic`.`subnet_id`, `nic`.`vnet_id`, `nic`.`device_number`, `nic`.`mac_address` from `nic` where `nic`.`id` in (select max(`nic`.`id`) from `nic` join `vnet` on `vnet`.`id` = `nic`.`vnet_id` join `instance` on `instance`.`id` = `nic`.`instance_id` where (`nic`.`account_id` = 5 and `nic`.`removed` is null and `instance`.`removed` is null and `instance`.`system_container` is null) group by `nic`.`vnet_id`) |
| 653 | cattle | localhost:59086 | cattle | Query   |  470 | Copying to tmp table | select `nic`.`id`, `nic`.`name`, `nic`.`account_id`, `nic`.`kind`, `nic`.`uuid`, `nic`.`description`, `nic`.`state`, `nic`.`created`, `nic`.`removed`, `nic`.`remove_time`, `nic`.`data`, `nic`.`instance_id`, `nic`.`network_id`, `nic`.`subnet_id`, `nic`.`vnet_id`, `nic`.`device_number`, `nic`.`mac_address` from `nic` wh
```

```sql
mysql>                     explain select `nic`.`id`, `nic`.`name`, `nic`.`account_id`, `nic`.`kind`, `nic`.`uuid`, `nic`.`description`, `nic`.`state`, `nic`.`created`, `nic`.`removed`, `nic`.`remove_time`, `nic`.`data`, `nic`.`instance_id`, `nic`.`network_id`, `nic`.`subnet_id`, `nic`.`vnet_id`, `nic`.`device_number`, `nic`.`mac_address` from `nic` where `nic`.`id` in (select max(`nic`.`id`) from `nic` join `vnet` on `vnet`.`id` = `nic`.`vnet_id` join `instance` on `instance`.`id` = `nic`.`instance_id` where (`nic`.`account_id` = 5 and `nic`.`removed` is null and `instance`.`removed` is null and `instance`.`system_container` is null) group by `nic`.`vnet_id`) 

+----+--------------------+----------+-------------+------------------------------------------------------------------------+------------------------------------+---------+------------------------+-------+---------------------------------------------------------------------------------------------------+
| id | select_type        | table    | type        | possible_keys                                                          | key                                | key_len | ref                    | rows  | Extra                                                                                             |
+----+--------------------+----------+-------------+------------------------------------------------------------------------+------------------------------------+---------+------------------------+-------+---------------------------------------------------------------------------------------------------+
|  1 | PRIMARY            | nic      | ALL         | NULL                                                                   | NULL                               | NULL    | NULL                   | 44881 | Using where                                                                                       |
|  2 | DEPENDENT SUBQUERY | nic      | index_merge | fk_nic__account_id,fk_nic__instance_id,fk_nic__vnet_id,idx_nic_removed | idx_nic_removed,fk_nic__account_id | 9,9     | NULL                   |   142 | Using intersect(idx_nic_removed,fk_nic__account_id); Using where; Using temporary; Using filesort |
|  2 | DEPENDENT SUBQUERY | vnet     | eq_ref      | PRIMARY                                                                | PRIMARY                            | 8       | cattle.nic.vnet_id     |     1 | Using index                                                                                       |
|  2 | DEPENDENT SUBQUERY | instance | eq_ref      | PRIMARY,idx_instance_removed                                           | PRIMARY                            | 8       | cattle.nic.instance_id |     1 | Using where                                                                                       |
+----+--------------------+----------+-------------+------------------------------------------------------------------------+------------------------------------+---------+------------------------+-------+---------------------------------------------------------------------------------------------------+
4 rows in set (0.00 sec)

mysql> select count(*) from cattle.nic;
+----------+
| count(*) |
+----------+
|    40005 |
+----------+
1 row in set (0.05 sec)
```

**Comment by samvignoli** [comment]
Fixed deleting old rows. Seems like a bug.

**Comment by ibuildthecloud** [comment]
@alena1108 and I are looking at it.  We just introduced that query and it seems quite bad....

**Comment by samvignoli** [comment]
Thank you! :clap:

**Comment by deniseschannon** [comment]
Fixed as of v0.50.0.
